### PR TITLE
Implement Service & Service Object for BigQuery

### PR DESCRIPTION
**Pull request by stephenplusplus**
For #904 

This hooks BigQuery up with Service & ServiceObject.

### CODE REVIEW

File: lib/bigquery/dataset.js [hunk 1]
```diff
@@ -46,20 +53,121 @@ var streamRouter = require('../common/stream-router.js');
...
+ * });
+ * var dataset = bigquery.dataset('institutions');
  */
 function Dataset(bigQuery, id) {
+  var methods = {
+    /**
+     * Create a dataset.
+     *
+     * @example
+     * dataset.create(function(err, dataset, apiResponse) {
```
Comment (callmehiphop) [inline lib/bigquery/dataset.js]: Do we want to document the params here? Or is this something that we won't really be advertising and it's just going to be masked behind `autoCreate`?

Comment (callmehiphop) [inline lib/bigquery/dataset.js]: Or does this get mixed in with the `ServiceObject` docs?

Comment (stephenplusplus) [inline lib/bigquery/dataset.js]: This method is exposed in the docs, but deliberately undocumented in this case, as `create` takes no arguments and having duplicating the callback argument docs is not ideal.

This is a very unique case, though, as most create methods will accept an `options` or `config` object, and we will show param docs that link to the original method (`parentClass.create____`) so the user can see the full list of docs.

**Comment by stephenplusplus** [comment]
Docs preview: http://stephenplusplus.github.io/gcloud-node/#/docs/master/bigquery

**Pull request merged**